MastodonSDK/Sources/MastodonCore/FetchedResultsController/FeedFetchedResultsController.swift

```diff
@@ -9,87 +9,86 @@
-            .map { objectIDs in objectIDs.map { ManagedObjectRecord(objectID: $0) } }
-            .assign(to: &$records)
-        
-        fetchedResultsController.delegate = self
-        
-        $predicate
-            .removeDuplicates()
-            .receive(on: DispatchQueue.main)
-            .sink { [weak self] predicate in
-                guard let self = self else { return }
-                self.fetchedResultsController.fetchRequest.predicate = predicate
-                do {
-                    try self.fetchedResultsController.performFetch()
-                } catch {
-                    assertionFailure(error.localizedDescription)
+    public func update(status: MastodonStatus) {
+        var newRecords = Array(records)
+        for (i, record) in newRecords.enumerated() {
+            if record.status?.id == status.id {
+                newRecords[i] = .fromStatus(status, kind: record.kind)
+            } else if let reblog = status.reblog, reblog.id == record.status?.id {
+                newRecords[i] = .fromStatus(status, kind: record.kind)
+            } else if let reblog = record.status?.reblog, reblog.id == status.id {
+                // Handle reblogged state
+                switch status.entity.reblogged {
+                case .some(true):
+                    newRecords[i] = .fromStatus({
+                        let stat = MastodonStatus.fromEntity(records[i].status!.entity)
+                        stat.isSensitiveToggled = status.isSensitiveToggled
+                        stat.reblog = .fromEntity(status.entity)
+                        return stat
+                    }(), kind: record.kind)
+                case .some(false), .none:
+                    newRecords[i] = .fromStatus({
+                        let stat = MastodonStatus.fromEntity(status.entity)
+                        stat.isSensitiveToggled = status.isSensitiveToggled
+                        return stat
+                    }(), kind: record.kind)
                 }
+
+            } else if let reblog = record.status?.reblog, reblog.id == status.reblog?.id {
+                // Handle re-reblogged state
+                newRecords[i] = .fromStatus(status, kind: record.kind)
             }
-            .store(in: &disposeBag)
+        }
+        records = newRecords
+    }
+    
+    public func delete(status: MastodonStatus) {
+        self.records.removeAll { $0.id == status.id }
     }
 }
 
-// MARK: - NSFetchedResultsControllerDelegate
-extension FeedFetchedResultsController: NSFetchedResultsControllerDelegate {
-    public func controller(
-        _ controller: NSFetchedResultsController<NSFetchRequestResult>,
-        didChangeContentWith snapshot: NSDiffableDataSourceSnapshotReference
-    ) {
-        let snapshot = snapshot as NSDiffableDataSourceSnapshot<String, NSManagedObjectID>
-        self._objectIDs.send(snapshot.itemIdentifiers)
+private extension FeedFetchedResultsController {
```

Comment:
I don't know yet if I'm convinced, but I kinda like this `private extension`-approach 👍

Reply:
We could also make it a private function inside the owning class, but this it's a bit more structure IMO.